src/Select/Select.js

```diff
@@ -0,0 +1,539 @@
+ * limitations under the License.
+ */
+
+/* @flow */
+import React, { Component } from 'react';
+import { findDOMNode } from 'react-dom';
+import scrollIntoViewIfNeeded from 'scroll-into-view-if-needed';
+import { generateId } from '../utils';
+import { createStyledComponent } from '../styles';
+import { createThemedComponent, mapComponentThemes } from '../themes';
+import Dropdown, {
+  componentTheme as dropdownComponentTheme
+} from '../Dropdown/Dropdown';
+import ItemMatcher from '../Dropdown/ItemMatcher';
+import { getItems } from '../Menu/Menu';
+import SelectTrigger, {
+  componentTheme as selectTriggerComponentTheme
+} from './SelectTrigger';
+import { pxToEm } from '../styles';
+
+import type { Items, ItemGroups } from '../Menu/Menu';
+import type { Item } from '../Menu/MenuItem';
+
+type Props = {
+  /**
+   * Data from which the [Menu](../menu#data) will be constructed
+   * (see [example](#data))
+   */
+  data: Items | ItemGroups,
+  /**
+   * Index of item to be highlighted upon initialization. Primarily for
+   * use with uncontrolled components.
+   */
+  defaultHighlightedIndex?: number,
+  /**
+   * Open the Select upon initialization. Primarily for use with uncontrolled
+   * components.
+   */
+  defaultIsOpen?: boolean,
+  /**
+   * Item selected upon initialization. Primarily for use with uncontrolled
+   * components.
+   */
+  defaultSelectedItem?: Item,
+  /** Disables the control */
+  disabled?: boolean,
+  /** Index of the highlighted item. For use with controlled components. */
+  highlightedIndex?: number,
+  /** Id of the Select */
+  id?: string,
+  /** Indicates that the value of the element is invalid */
+  invalid?: boolean,
+  /** Determines whether the Select is open. For use with controlled components. */
+  isOpen?: boolean,
+  /**
+   * Plugins that are used to alter behavior. See
+   * [PopperJS docs](https://popper.js.org/popper-documentation.html#modifiers)
+   * for options.
+   */
+  modifiers?: Object,
+  /** Name of the field when submitted in a form */
+  name?: string,
+  /**
+   * Called when an item is selected and it is different than the previously
+   * selected item.
+   */
+  onChange?: (item: Item, event: SyntheticEvent<>) => void,
+  /** Called when Select is closed */
+  onClose?: (event: SyntheticEvent<>) => void,
+  /** Called when Select is opened */
+  onOpen?: (event: SyntheticEvent<>) => void,
+  /** Called when an item is selected */
+  onSelect?: (item: Item, event: SyntheticEvent<>) => void,
+  /** Text displayed when there is no item selected */
+  placeholder?: string,
+  /** Placement of the Select */
+  placement?: 'bottom-end' | 'bottom-start' | 'top-end' | 'top-start',
+  /** Indicates that the user cannot modify the value of the control */
+  readOnly?: boolean,
+  /** Indicates that the user must select a value before submitting a form */
+  required?: boolean,
+  /** The selected item. For use with controlled components. */
+  selectedItem?: Item,
```

Comment:
It seems inconsistent for highlightedIndex to use an index and selectedItem to use an Item.

Reply:
Good point, need to think about that one a bit...

Reply:
It is probably okay, since these are really only control props. They just mirror the state.

Reply:
We need to read values off selected item, but not the highlighted item.